main.go

```diff
@@ -1,83 +1,47 @@
 package main
 
 import (
-	"bufio"
 	"fmt"
-	"os"
+	"reflect"
 )
 
-// Data structure = linked list as a stack [data+head]->[2]->[1]->[0]
-
-// acts as head of list & contains DB
-type HeadDB struct {
-	// access to stack of operations
-	current *Node
-	// most recent ops for same key
-	data map[string]string
-}
-
 // represents transaction with a hash of operations
 type Node struct {
 	parent *Node
-	ops    map[string]struct
-}
-
-type Read struct {
-}
-
-type Write struct {
-	value string
-}
-
-type Delete struct {
+	data   map[string]string
 }
 
-// might need an interface for operations (?)
-
-func (h *HeadDB) WriteTo(k,v string) *HeadDB {
-	// reference new Write Struct
-	writeOp := Write{value: v}
-	// access current ops list
-	list = &h.current.ops
-	*list[k] = writeOp
+type Root struct {
+	head *Node
 }
 
-func (h *HeadDB) Start() *HeadDB {
-	newOps := make(map[string]Op)
-	// newNode where parent is current
-	newNode := &Node{parent: h.current ops: newOps }
-	// Then reset Head's current to new node
-	h.current = newNode
-	return h
+// start of REPL: make first Node and set as head
+func initialize() *Root {
+	m := make(map[string]string)
+	newNode := &Node{parent: nil, data: m}
+	return &Root{head: newNode}
 }
 
-func (h *HeadDB) Abort() *HeadDB {
-	// TODO: move the head pointer and delete current Node
-	return h
+// store k,v to current node's data
+func (n *Node) Write(k, v string) string {
+	d := n.data
+	d[k] = v
+	return ""
 }
 
-func (h *HeadDB) Commit() *HeadDB {
-	// TODO: write current commands to datastore
-	return h
+func (n *Node) Read(k string) string {
+	d := n.data
+	return d[k]
```

Comment:
what is the expected behavior if the key does not exist? (i just don't remember what the spec says). make sure you're handling it.